numpy/_core/numeric.py

```diff
@@ -685,12 +685,12 @@ def flatnonzero(a):
     return np.nonzero(np.ravel(a))[0]
 
 
-def _correlate_dispatcher(a, v, mode=None):
-    return (a, v)
+def _correlate_dispatcher(a, v, mode=None, out=None):
+    return (a, v, out)
 
 
 @array_function_dispatch(_correlate_dispatcher)
-def correlate(a, v, mode='valid'):
+def correlate(a, v, mode='valid', out=None):
```

Comment:
```suggestion
def correlate(a, v, mode='valid', *, out=None):
```
Also in the other, we should force kwarg use unless there is a clear reasoning why non-kwarg usage is convenient and clear.